guide/14-deep-learning/FCN_MLModel_guide-final_draft.ipynb

```diff
@@ -0,0 +1,517 @@
+{
```

Comment:
<p>Introduction</p><p> </p><blockquote>The FCN and MLModel frameworks are recently added modules in arcgis.learn that can handle data preparation and model tabular data using deep learning and machine learning algorithms respectively. These models can be trained with features and features with Rasters data while inferencing can happen on feature data, feature with Rasters, or just rasters. The two methodology for modelling are as follows:</blockquote><blockquote><br></blockquote><blockquote>1. Fully Connected Network(FCN) – This is a deep learning framework called FullyConnectedNetwork available in the arcgis.learn module in ArcGis API for Python, and is based on TabularLearner from FastAI</blockquote><blockquote><br></blockquote><blockquote>2. Machine Learning Model(MLModel) - In the second option any machine learning algorithm from scikit learn can be implemented via the MLModel framework available in arcgis.learn. This framework can deploy any ML algorithm from the scikit learn library just by passing the name of the algorithm and its relevant parameters as keyword arguments. This tool makes it possible for model selection utlilizing the several machine learning models available in scikit learn by comparing the model accuracies.</blockquote><p> </p><p>The <code>arcgis.learn</code> module includes <a href="https://developers.arcgis.com/python/api-reference/arcgis.learn.html#fullyconnectednetwork" target="_blank">FullyConnectedNetwork</a> and <a href="https://developers.arcgis.com/python/api-reference/arcgis.learn.html#mlmodel" target="_blank">MLModel</a> to train machine learning and deep learning models on tabular data. Tabular data can be any vector data in the form of a <a href="https://developers.arcgis.com/python/guide/working-with-feature-layers-and-features/" target="_blank">feature&nbsp;layer</a>&nbsp;or&nbsp;<a href="https://developers.arcgis.com/python/guide/introduction-to-the-spatially-enabled-dataframe/" target="_blank">spatially&nbsp;enabled&nbsp;dataframe</a>. Explanatory variables can also include data from <a href="https://developers.arcgis.com/python/guide/using-imagery-layers/" target="_blank">raster</a> layers.</p><p> </p><p>FullyConnectedNetwork model is based on deep neural networks whereas MLModel uses machine learning algorithms in order to train a model.</p><p> </p><p>The MLModel allows you to use any <a href="https://scikit-learn.org/stable/supervised_learning.html#supervised-learning" target="_blank">regression</a> or <a href="https://scikit-learn.org/stable/supervised_learning.html#supervised-learning" target="_blank">classification</a> model from scikit learn whereas FullyConnectedNetwork allows you to train a fully trained deep neural network with a specified number of layers.</p><p>&nbsp;</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/718/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>